test/mpi/main.cc

```diff
@@ -236,9 +268,41 @@ static void process_root(const char *arg)
     PrintHelp();
 }
 
+int init_rand_seed(int user_seed)
+{
+    int rank, seed;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    if (0 > user_seed) {
+        if (0 == rank) {
+            seed = time(NULL) % 32768;
+        }
+    } else {
+        seed = user_seed;
+    }
+    MPI_Bcast(&seed, 1, MPI_INT, 0, MPI_COMM_WORLD);
+    if (0 != rank) {
+        seed += rank;
+    }
+    return seed;
+}
+
+void PrintInfo()
+{
+    int world_rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
+
+    if (world_rank) {
+        return;
+    }
+    std::cout << "\n===== UCC MPI TEST INFO =======\n"
+              << "   seed        : " << std::to_string(test_rand_seed) << "\n"
+              <<   "===============================\n"
+              << std::endl;
+}
+
 void ProcessArgs(int argc, char** argv)
 {
-    const char* const short_opts = "c:t:m:d:o:M:I:r:h";
+    const char* const short_opts = "c:t:m:d:o:M:I:r:s:C:D:Z:h";
```

Comment:
need to check somewhere that requested memory type is available in ucc, perhaps we can do it in mtype_str_to_type

Reply:
fixed